### ISSUE TITLE
[bug] CoreDNS Dashboard No Data

### ISSUE THREAD

**Issue by felipewnp** [dotdc/grafana-dashboards-kubernetes#70]
### Describe the bug

Hi, and thanks for the good set of Dashboards @dotdc !

I'm having some trouble with the CoreDNS dashboard.

Several graphs and statuses don't show any data, displaying the "No Data" placeholder.

I've noticed that the filter for CoreDNS is a job and not a pod.

At least in my EKS, the CoreDNS is a daemonset and not a job.

![image](https://github.com/dotdc/grafana-dashboards-kubernetes/assets/34146218/c8153f7f-8ef5-4392-8791-9a41665131a4)

Is there something I could do or change?

Thanks =D !

**Comment by dotdc** [comment]
Hi @felipewnp,
Do you have any metrics from the broken panels?
If yes, could you share sample data for one or two?

**Comment by felipewnp** [comment]
> Hi @felipewnp, Do you have any metrics from the broken panels? If yes, could you share sample data for one or two?

I've found that the job name in the Prometheus metrics is not coredns, but _**kube-prometheus-coredns**_.

Not sure why.

When I changed the Dashboard JSON to kube-prometheus-coredns, it started working.

What is supposed to be the default value?

**Comment by dotdc** [comment]
@felipewnp 

For the job, you could rename it using a relabeilling rule, but the job label is only used by:
- CoreDNS - Health Status
- CoreDNS - CPU Usage by instance
- CoreDNS - Memory Usage by instance

This doesn't explain the `no data` on the other panels.

Could you share:
- setup information (k8s version, kube-prometheus-stack version...)?
- sample data from `coredns_dns_requests_total` in Grafana explore for example?

Data example: 
```
coredns_dns_requests_total{container="coredns",endpoint="http-metrics",family="1",instance="10.42.2.186:9153",job="coredns",namespace="kube-system",pod="coredns-5d78c9869d-2lc7d",proto="udp",server="dns://:53",service="kube-prometheus-stack-coredns",type="A",zone="."}
```

**Comment by rcattin** [comment]
Hello
I had the same issue,
The job label is also used in the $instance variable that is used in all the other panels

**Comment by Haleygo** [comment]
@dotdc Hey,
>but the job label is only used by:
>CoreDNS - Health Status
CoreDNS - CPU Usage by instance
CoreDNS - Memory Usage by instance

Is this on purpose?
What do you think about replacing job with $job, and add [ad-hoc filter](https://grafana.com/docs/grafana/latest/dashboards/variables/add-template-variables/#add-ad-hoc-filters) for them?

**Comment by AndrisJrs** [comment]
I had same issue on "standard" Prometheus. Moving to Kube Prometheus stack fixed the issue. Probably Core DNS exporter is missing.

**Comment by dotdc** [comment]
Sorry for the delay.
Just made a PR with a potential fix : https://github.com/dotdc/grafana-dashboards-kubernetes/pull/77

Let me know if this solves the issue on your side.

cc @felipewnp @rcattin @Haleygo

**Comment by felipewnp** [comment]
Hi @dotdc Also sorry for the delay.

It fixed a lot of the items in the CoreDNS dashboard, but these 3 are still broken:

<details>

![image](https://github.com/dotdc/grafana-dashboards-kubernetes/assets/34146218/57f84d8d-b522-4593-bb49-a8032803ebd4)

</details>

I think it's because the job filter in these graph items are hardcoded to "coredns" and not to the $job variable:

<details>

![image](https://github.com/dotdc/grafana-dashboards-kubernetes/assets/34146218/9ba505be-0511-4be1-885a-93a530589186)

</details>

Should/Can I make a PR for this change?

**Comment by dotdc** [comment]
Hi @felipewnp,
You're right, sorry about that.
Of course, you can make a PR, please also bump the version of the dashboard.

**Comment by LeoAnt02** [comment]
Hi @dotdc 

I'me facing the same issue on my end 
Not sure what was the solution ?

I'm still new to grafana and prometheus

**Comment by dotdc** [comment]
> Hi @dotdc
> 
> I'me facing the same issue on my end Not sure what was the solution ?
> 
> I'm still new to grafana and prometheus

This should be fixed by #89
Can you try with the latest version?
